cloudinit/distros/__init__.py

```diff
@@ -274,6 +279,32 @@ def _write_network(self, settings):
             "_write_network_config needs implementation.\n" % self.name
         )
 
+    @property
+    def dhcp_client(self) -> dhcp.DhcpClient:
+        """access the distro's preferred dhcp client
+
+        if no client has been selected yet select one - uses
+        self.dhcp_client_priority, which may be overriden in each distro's
+        object to eliminate checking for clients which will not be provided
+        by the distro
+        """
+        if self._dhcp_client:
+            return self._dhcp_client
+        self._select_dhcp_client()
+        return self._dhcp_client
```

Comment:
Your proposal makes sense to have the private attribute check in only one place to avoid such duplication. And since there are no callers of _select_dhcp_client it makes sense to drop it altogether  thank you for that.